ui/src/fragments/IconButton.svelte

```diff
@@ -1,26 +1,40 @@
 <script>
   import { LoadingIcon } from '@fragments';
+  import {
+    state,
+  } from '@root/state';
   export let loading = false;
   export let async = false;
   export let disabled = false;
   export let tooltip = '';
   export let transparent = false;
-  export let noHover = false;
-  export let whiteIcon = true;
+  export let changeColorOnHover = false;
+  export let whiteIcon;
   export let active;
   export let icon;
-  console.log("testy")
-  console.log(whiteIcon)
 </script>
 
 <button
-  class="py-2 px-2 flex items-center gap-4 hover:duration-500 overflow-hidden rounded-lg dark:fill-white"
+  class="py-2 px-2 flex items-center gap-4 hover:duration-500 overflow-hidden rounded-lg stroke-grey"
   class:pointer-events-none={disabled}
   class:opacity-50={disabled}
   class:cursor-not-allowed={disabled}
-  class:bg-hover={active}
-  class:bg-panels={!transparent && !active}
-  class:hover:bg-hover={!noHover}
+  class:bg-hover={active && !$state.darkmode && !changeColorOnHover}
+  class:bg-panels={!transparent && !active && !$state.darkmode}
+  class:fill-grey={(!whiteIcon && !active) || icon.name.includes('People') && !changeColorOnHover}
+  class:fill-black={icon.name.includes('Chat') && active}
+  class:fill-white={whiteIcon && icon.name.includes('Moon')}
+  class:dark:fill-white={(icon.name.includes('Chat') && active) || whiteIcon}
+  class:stroke-grey={icon.name.includes('Like')}
+  class:hover:stroke-black={icon.name.includes('Like')}
+  class:dark:hover:stroke-white={icon.name.includes('Like')}
+  class:hover:bg-hover={!changeColorOnHover && !$state.darkmode}
+  class:hover:fill-black={changeColorOnHover && !$state.darkmode && !whiteIcon}
+  class:hover:fill-offwhite={changeColorOnHover && !$state.darkmode && (whiteIcon || $state.darkmode)}
+  class:dark:hover:fill-white={changeColorOnHover && $state.darkmode}
+  class:dark:hover:border-[#FFFFFF]={!changeColorOnHover && $state.darkmode}
+  class:border-[#FFFFFF]={active && $state.darkmode && !changeColorOnHover}
+  class:dark:border={$state.darkmode && !changeColorOnHover}
```

Comment:
Brady I am going to vomit, there's no way I am going to merge this. What are you trying to do here?

Reply:
moved most classes into invocation